### DISCUSSION TITLE
Require JDK 21+?

### DISCUSSION THREAD

**Discussion by bishabosha** [scala/toolkit#50] [Ideas]
So far in toolkit we are mostly recommending a blocking style. Are we any where near providing a good experience in toolkit with full support for virtual threads? - it seems the jdk http client (provided with sttp) is ready for virtual threads - and next we are considering web servers / database clients

**Comment by bishabosha** [comment]
also if we do require this, then is there a good experience in tooling when a dependency requires a minimum JDK? - if not can we specialise for toolkit?

**Reply by adpi2** [reply]
That's probably something we can specialize in Scala CLI, maybe not in other build tools though.

**Comment by adpi2** [comment]
Yes, I think it would make sense to require JDK 21. We are still in experimental phase, so by the time we release a stable version, JDK 21 will be more widespread.

Also the Toolkit is mostly made for scripting and prototyping. If anyone needs more compatibility they can still pick parts of the Toolkit, and older versions of the libraries. The Toolkit is more about what we recommend as the default for the masses.
